### DISCUSSION TITLE
Sugestão: Sistema de Addon e outros pensamentos.

### DISCUSSION THREAD

**Discussion by neofln** [RamonSilva20/mapos#1640] [Ideas]
+Sou fã do Mapos e venho utilizando a um bom tempo, já contribui financeiramente algumas vezes com os programadores, muitas vezes vi o pessoal sofrendo pra adicionar as PR, e algumas nas quais eu tinha interesse foram recusadas em virtude de manter o projeto funcional para uma gama maior de utilizadores imagino. 

Como as necessidades de utilizadores diferentes são diferentes, um sistema de Addons resolveria a encrenca, quem quisesse poderia adicionar funcionalidades adicionais escritas por terceiros e verificadas (ou não) quanto a segurança, pelo pessoal do MapOS, poderia haver um repositório com lista e explicação de cada um, algo parecido com o que fazíamos muito antigamente no Mirc... 

Evidentemente eu entendo que por ser um programa de código livre qualquer um pode a qualquer tempo modificar os fontes e fazer dele o que bem quer, no entanto fica difícil muitas vezes compartilhar estas mudanças, e até atualizar o Map-OS posteriormente em virtude das PR.

Acredito que com um sistema de Addon oficial, quem quisesse compartilhar melhorias poderia fazê-lo mais facilmente, e claro os Addons cujo o download e o uso por parte da comunidade do Map-OS fossem recorrentes poderiam ser incorporados ao sistema base em versões futuras.

Um sistema como esse modularizaria o Map-OS, colocando ele um nível acima do que a maioria faz por aí.
Acredito que desta forma o sistema melhoraria/evoluiria com mais frequência sem a necessidade de instalar do zero uma nova versão e importar banco de dados de novo, corrigir as falhas comuns de emitente, etc...

+Implementar as configurações de e-mail, no menu lateral e-mail, e um sub menu para exibir os e-mails enviados.

+Centralizar todas as configurações necessárias em um mesmo arquivo e preferencialmente adicionar interface gráfica para configurar todas elas.

Foi o que me veio a mente neste momento... ;D

Mais uma vez reitero meu agradecimento a todos vocês que fazem do Map-OS um sistema ainda melhor a cada versão.

Abraços.

**Comment by willph** [comment]
Obrigado por compartilhar conosco seu ponto de vista.
Bom final de semana.

**Comment by Pr3d4dor** [comment]
@neofln Esse sistema de "addons" é interessante e eu concordo com os seus pontos. A única questão é que não é tão fácil de ser implementado e mantido.

Eu acho que para o mapos atual em CodeIgniter é melhor deixar do jeito que está. Eu voltei a trabalhar no projeto Mapos-Laravel e acho que seria uma boa adição pra ele, porque seria mais fácil de manter do que o código atual. Porém no Mapos-Laravel ainda para a versão 1 o ideal é apenas refazer o mapos do estado atual e ter a possibilidade de migrar do mapos antigo para ele.

**Reply by neofln** [reply]
Eu entendo as dificuldades, principalmente por ser um projeto que é desenvolvido gratuitamente.
Só quis mesmo compartilhar a ideia da modularização, minha ideia é sempre ajudar, não complicar a vida de ninguém hehehe ;D
Tomara que em algum momento seja possível implementar, quem sabe em alguma versão futura do Map-OS-Laravel.

**Reply by neofln** [reply]
Lastimo muito que a vida não me proporcione tempo para me envolver mais no projeto mesmo que financeiramente no momento, acredito muito no Map-OS como solução de código livre para ajudar no fomento dos pequenos negócios.
Me pergunto, será que não haveria forma de receber algum incentivo governamental para tocar o projeto?
Abraços.

**Reply by willph** [reply]
Eu pelo menos não sei se tem como conseguir recursos para focar apenas nele. 😁

Opiniões sempre serão bem vindas.
Obrigado mesmo assim.

Ótima tarde

**Comment by Edivandro** [comment]
Falando em Frameworks . . .

Alguém conhece o Adianti Framework ?  Framework BRASILEIRO open com material educativo muito bom.
Ele parece facilitar para o desenvolvimento modular, feito pensando nesses detalhes.

Ele parece ser uma ótima pedida para desenvolver sistemas em PHP, ele conta com 'template de desenvolvimento' ou seja um grande gabarito para criação de novos projetos que possui controle de login e permissões de acesso por usuários, grupos de usuários e programas. Possui controles multi unidade, e multi idioma. Oferece gestão e compartilhamento de documentos entre usuários, mensagens entre usuários, e notificações. Possui controles nativos de auditoria com logs de acesso, logs de mudança de registros, logs de SQL, los de requisição HTTP, REST, dentre outros.

Se vocês conhecem esse framework qual a opnião de vocês  sobre ele e a possibilidade de migrar ???
